packages/core/admin/server/services/permission/permissions-manager/query-builers.js

```diff
@@ -3,79 +3,69 @@
-  common: VALID_REST_OPERATORS.map(op => `$${op}`),
-  boolean: ['$or', '$and'],
-  cleanable: ['$elemMatch'],
+const operatorsMap = {
+  $in: '$in',
+  $nin: '$notIn',
+  $exists: '$notNull',
+  $gte: '$gte',
+  $gt: '$gt',
+  $lte: '$lte',
+  $lt: '$lt',
+  $eq: '$eq',
+  $ne: '$ne',
+  $and: '$and',
+  $or: '$or',
+  $not: '$not',
+};
+
+const mapKey = key => {
+  if (_.isString(key) && key.startsWith('$') && key in operatorsMap) {
+    return operatorsMap[key];
+  }
+  return key;
 };
 
 const buildCaslQuery = (ability, action, model) => {
   const query = rulesToQuery(ability, action, model, o => o.conditions);
-  return _.get(query, '$or[0].$and', {});
+  return _.get(query, '$or[0]', {});
 };
 
 const buildStrapiQuery = caslQuery => {
-  const transform = _.flow([flattenDeep, cleanupUnwantedProperties]);
-  return transform(caslQuery);
-};
-
-const flattenDeep = condition => {
-  if (_.isArray(condition)) {
-    return _.map(condition, flattenDeep);
-  }
-
-  if (!_.isObject(condition)) {
-    return condition;
-  }
-
-  const shouldIgnore = e => !!ops.common.includes(e);
-  const shouldPerformTransformation = (v, k) => _.isObject(v) && !_.isArray(v) && !shouldIgnore(k);
-
-  const result = {};
-  const set = (key, value) => (result[key] = value);
-
-  const getTransformParams = (prevKey, v, k) =>
-    shouldIgnore(k) ? [`${prevKey}_${k.replace('$', '')}`, v] : [`${prevKey}.${k}`, v];
-
-  _.each(condition, (value, key) => {
-    if (ops.boolean.includes(key)) {
-      set(key.replace('$', '_'), _.map(value, flattenDeep));
-    } else if (shouldPerformTransformation(value, key)) {
-      _.each(flattenDeep(value), (v, k) => {
-        set(...getTransformParams(key, v, k));
-      });
-    } else {
-      set(key, flattenDeep(value));
-    }
-  });
-
-  return result;
+  const transform = _.flow([unwrapDeep]);
+  const res = transform(caslQuery);
+  return res;
 };
 
-const cleanupUnwantedProperties = condition => {
-  if (!_.isObject(condition)) {
-    return condition;
+const unwrapDeep = obj => {
+  if (!_.isPlainObject(obj) && !_.isArray(obj)) {
+    return obj;
   }
-
-  if (_.isArray(condition)) {
-    return condition.map(cleanupUnwantedProperties);
+  if (_.isArray(obj)) {
+    return obj.map(v => unwrapDeep(v));
   }
 
-  const shouldClean = e =>
-    typeof e === 'string' ? ops.cleanable.find(o => e.includes(`.${o}`)) : undefined;
-
   return _.reduce(
-    condition,
-    (acc, value, key) => {
-      const keyToClean = shouldClean(key);
-      const newKey = keyToClean ? key.split(`.${keyToClean}`).join('') : key;
-
-      return {
-        ...acc,
-        [newKey]: _.isArray(value) ? value.map(cleanupUnwantedProperties) : value,
-      };
+    obj,
+    (acc, v, k) => {
+      const key = mapKey(k);
+
+      if (_.isPlainObject(v)) {
+        if ('$elemMatch' in v) {
+          v = v.$elemMatch; // removing this key
+        }
+        _.setWith(acc, key, unwrapDeep(v));
+      } else if (_.isArray(v)) {
+        _.setWith(
```

Comment:
oneline ?

Reply:
it's prettier :'( Because of the arrow function I think